abi/src/main/java/org/web3j/abi/TypeReference.java

```diff
@@ -16,6 +25,7 @@
  */
 public abstract class TypeReference<T extends org.web3j.abi.datatypes.Type>
         implements Comparable<TypeReference<T>> {
+    protected static Pattern ARRAY_SUFFIX = Pattern.compile("\\[(\\d*)\\]$");
```

Comment:
IntelliJ reports a warning: `Redundant character scape \\`

Reply:
hmmm that regex looks correct to me and IntelliJ shows error with "\[(\\d*)\]$" and "\\[(\d*)\\]$". I think you need "\\" there to created the regex string `\[(\d*)\]$`.

Reply:
`\[(\d*)]$` does not show any error for me and removing `\\` removes the warning.

Reply:
hmmm that looks correct to me because we want to construct the regex string `\[(\d*)\]$`. What do you suggest?

Reply:
ok, if that works. I guess ] without [ is interpreted literally.